Source/WebKit/WebProcess/Extensions/API/WebExtensionAPISidebarAction.h

```diff
@@ -35,6 +35,33 @@ OBJC_CLASS NSString;
 
 namespace WebKit {
 
+using SidebarError = NSString *;
+// In this variant, `monostate` indicates that we have neither a window or tab identifier, but no error
+using ParseResult = std::variant<std::monostate, WebExtensionTabIdentifier, WebExtensionWindowIdentifier, SidebarError>;
+
+template<typename T, typename VARIANT_T>
+struct isVariantMember;
+template<typename T, typename... ALL_T>
+struct isVariantMember<T, std::variant<ALL_T...>> : public std::disjunction<std::is_same<T, ALL_T>...> { };
+
+template<typename OptType, typename... Types>
+std::optional<OptType> variantTypeToOptional(std::variant<Types...>& variant)
+{
+    if (std::holds_alternative<OptType>(variant))
+        return WTFMove(std::get<OptType>(variant));
+    return std::nullopt;
+}
+
+template<typename VariantType>
+NSString *indicatesError(const VariantType& variant)
```

Comment:
Can this move to the `mm` file?

Reply:
I don't think it can, since it's templated and I'm using it in multiple .mm files